### DISCUSSION TITLE
Set tabBarBadge for other Tab of a Bottom Tab Navigator

### DISCUSSION THREAD

**Discussion by GaylordP** [react-navigation/react-navigation#12343] [Q&A]
Hello and thanks you very much for your library :)

I know it’s possible to set badges from a screen ([documentation](https://reactnavigation.org/docs/customizing-tabbar/#add-badges-to-icons)) using:

```
navigation.setOptions({
    tabBarBadge: unreadMessagesCount,
});
```

I also know it’s possible to navigate through the React Navigation tree, for example, to set the badge in a Bottom Tab Navigator:

```
navigation.getParent().setOptions({
    tabBarBadge: unreadMessagesCount,
});
```

However, if my parent is a Bottom Tab Navigator within a Native Stack Navigator, is it possible to select which tab the badge will be set on?

For example, something like this to apply the badge to the second tab of the Bottom Tab Navigator:

```
navigation.getParent().getParent().get(2).setOptions({
    tabBarBadge: unreadMessagesCount,
});
```

Thanks very much :)

**Comment by satya164** [comment]
It's not possible to set another tab's options from a different tab. You'd need to have some kind of event listener or global state management setup to notify that tab about the value and then that tab call `setOptions`.

Also, instead of `getParent().getParent()` I'd recommend using IDs https://reactnavigation.org/docs/navigation-object/#getparent
